solr/core/src/java/org/apache/solr/rest/RestManager.java

```diff
@@ -468,16 +468,17 @@ protected void onManagedDataLoadedFromStorage(NamedList<?> managedInitArgs, Obje
     @SuppressWarnings("unchecked")
     @Override
     public synchronized void doPut(BaseSolrResource endpoint, Object json) {
-      if (json instanceof Map) {
+      if ((json instanceof Map)
+          && ((Map<?, ?>) json).size() == 1
```

Comment:
Not sure about adding these conditions, maybe I would let the Exception come later on in the process as was before. Even if probably with a less clear message.
Let's see what @alessandrobenedetti thinks about it.

Reply:
Yes, I am not sure as well. Otherwise, we could keep the if statement like it is and enrich only the exception message, for example: 
```
"Expected Map to create a new ManagedResource but received a "
                + json.getClass().getName() + ". If one is trying to put the payload, check for a typo in the endpoint or create the new ManagedResource before pushing data"
```

Reply:
I wouldn't add the condition, to be honest, I am not sure if anything else ends up there that shouldn't, we are forcing casting as well, it's not super clear what happens and the reasons when reading the code, furthermore it's not that clean as an exception anyway, I recommend we discard this and potentially in the future address the whole problem of malformed URL